### PR TITLE
videotoimage

### PR DESCRIPTION
**Pull request by shroffshivangi**
video to image code and read me

### CODE REVIEW

File: VideoToImage/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Chops video into images
```
Comment (jkurz) [inline VideoToImage/README.md]: can you format this to have some headers and maybe some bullet points? Markdown is better used if you do some simple formatting.

File: VideoToImage/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+It downloads a file from s3 and uploads it back to s3 after chopping the images locally.
+It saves the chopped image in the current working directory locally.
+
+The FRAMESPERSEC environment variable can be used to define the number of image shots you want per second.
+
+Install ffmpeg into the console before running the code.
+Following command installs ffmpeg:
+```brew install ffmpeg --with-fdk-aac --with-ffplay --with-freetype --with-frei0r --with-libass --with-libvo-aacenc --with-libvorbis --with-libvpx --with-opencore-amr --with-openjpeg --with-opus --with-rtmpdump --with-schroedinger --with-speex --with-theora --with-tools```
+
+Finally run the code using: `python videotoimage.py`
```
Comment (jkurz) [inline VideoToImage/README.md]: what environment variables do I need to run this?

File: VideoToImage/videotoimage.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import botocore
+import subprocess as sp
+import os
+import boto3
+import datetime
+
+#No. of frames per second
+FRAMES = int(os.environ.get('FRAMESPERSEC'))
```
Comment (jkurz) [inline VideoToImage/videotoimage.py]: can you set a default of 1?

File: VideoToImage/videotoimage.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+def convertVideoToImage(srcKey, srcBucket):
+    s3 = boto3.resource('s3')
+    localFilename = '/tmp/{}'.format(os.path.basename(srcKey))
```
Comment (jkurz) [inline VideoToImage/videotoimage.py]: the srckey is really a filepath. Are you sure this works if the key is something `video/foo/bar/video.mp4`

File: VideoToImage/videotoimage.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    #Download video to local
+    try:
+        s3.Bucket(srcBucket).download_file(srcKey, localFilename)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == "404":
+            print("The object does not exist.")
```
Comment (jkurz) [inline VideoToImage/videotoimage.py:21]: you should exit here and not continue.

File: VideoToImage/videotoimage.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        else:
+            raise
+    #Chop video into images
+    cmd='ffmpeg -i '+str(localFilename)+' -r '+str(FRAMES)+' img_%04d.jpg'
+    sp.call(cmd,shell=True)
```
Comment (jkurz) [inline VideoToImage/videotoimage.py]: should probably do a try catch here. Just to make sure this doesn't fail and we catch errors from the shell process.

File: VideoToImage/videotoimage.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    #Upload file to s3
+    for imgFile in os.listdir("./"):
```
Comment (jkurz) [inline VideoToImage/videotoimage.py]: so we are writing to the local folder? Might want to write these to tmp also in an /tmp/output folder or something like that. Also make this configurable.

File: VideoToImage/videotoimage.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        if imgFile.find("img") != -1:
+            with open(imgFile, "rb") as imageFile:
+                f = imageFile.read()
+                imgByteArr = bytearray(f)
+                destination = 'video-image/'+str(datetime.datetime.now())+str(imgFile)
```
Comment (jkurz) [inline VideoToImage/videotoimage.py]: the path should be more dynamic and come from the `src_key`. whatever the path that the video came from, we should upload the images to the same folder or at least closely resemble the same file structure that the video came from. That way we can tell the program where to save it, by specifying the path from the video's path.

**Review by jkurz — COMMENTED** [review]
A few comments. Also add a .gitignore and ignore the DS_Store file.

File: VideoToImage/.gitignore [hunk 9]
```diff
@@ -0,0 +1 @@
+.DS_Store
```
Comment (jkurz) [inline VideoToImage/.gitignore]: you should just have to add one gitignore in the root for the files you want to globally ignore.

File: VideoToImage/videotoimage.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    tmpKey = srcKey.rsplit(".",1)
+    #Removing the video extention
+    destKey = tmpKey[0]
+
+    for imgFile in os.listdir(OUTPUT):
+        if imgFile.find("img") != -1:
+            with open(OUTPUT+imgFile, "rb") as imageFile:
+                f = imageFile.read()
+                imgByteArr = bytearray(f)
+                destination = destKey+'/'+str(datetime.datetime.now())+str(imgFile)
```
Comment (jkurz) [inline VideoToImage/videotoimage.py:41]: we don't need the current time in the file name. We just need the time of the video the image was taken from. That should already be there from ffmpg.

Comment (shroffshivangi) [reply VideoToImage/videotoimage.py:41]: I have added that to make the name unique.

**Comment by jkurz** [comment]
closing in favor of #2

**Pull request closed without merging**